sdk/keyvault/Azure.Security.KeyVault.Keys/Readme.md

```diff
@@ -0,0 +1,282 @@
+
+Console.WriteLine(key.Name);
+Console.WriteLine(key.KeyMaterial.KeyType);
+
+// Create a software RSA key
+var rsaCreateKey = new RsaKeyCreateOptions("rsa-key-name", hsm: false);
+Key rsaKey = Client.CreateRsaKey(rsaCreateKey);
+
+Console.WriteLine(rsaKey.Name);
+Console.WriteLine(rsaKey.KeyMaterial.KeyType);
+
+// Create a hardware Elliptic Curve key
+var echsmkey = new EcKeyCreateOptions("ec-key-name", hsm: true);
+Key ecKey = Client.CreateEcKey(echsmkey);
+
+Console.WriteLine(ecKey.Name);
+Console.WriteLine(ecKey.KeyMaterial.KeyType);
+```
+
+### Retrieve a Key
+`GetKey` retrieves a key previously stored in the Key Vault.
+
+```c#
+Key key = client.GetKey("key-name");
+
+Console.WriteLine(key.Name);
+Console.WriteLine(key.KeyMaterial.KeyType);
+```
+
+### Update an existing Key
+`UpdateKey` updates a key previously stored in the Key Vault.
+
+```c#
+Key key = client.CreateKey("key-name", KeyType.EllipticCurve);
+
+// You can specify additional application-specific metadata in the form of tags.
+key.Tags["foo"] = "updated tag";
+
+KeyBase updatedKey = client.UpdateKey(key);
+
+Console.WriteLine(updatedKey.Name);
+Console.WriteLine(updatedKey.Version);
+Console.WriteLine(updatedKey.Updated);
+```
+
+### Delete a Key
+`DeleteKey` deletes a key previously stored in the Key Vault. When [soft-delete][soft_delete] is not enabled for the Key Vault, this operation permanently deletes the key.
+
+```c#
+DeletedKey key = client.DeleteKey("key-name");
+
+Console.WriteLine(key.Name);
+Console.WriteLine(key.DeletedDate);
+```
+
+### List Keys
+This example lists all the keys in the specified Key Vault.
+
+```c#
+List<Response<KeyBase>> allKeys = Client.GetKeys();
+
+  foreach (Key key in allKeys)
+  {
+    Console.WriteLine(key.Name);
+  }
+```
+
+### Async create a Key
+Async APIs are identical to synchronous API. Note that all methods end with `Async`.
+
+```c#
+// Create a key of any type
+Key key = await client.CreateKeyAsync("key-name", KeyType.EllipticCurve);
+
+Console.WriteLine(key.Name);
+Console.WriteLine(key.KeyMaterial.KeyType);
+
+// Create a software RSA key
+var rsaCreateKey = new RsaKeyCreateOptions("rsa-key-name", hsm: false);
+Key rsaKey = await Client.CreateRsaKeyAsync(rsarsaCreateKeyKey);
+
+Console.WriteLine(rsaKey.Name);
+Console.WriteLine(rsaKey.KeyMaterial.KeyType);
+
+// Create a hardware Elliptic Curve key
+var echsmkey = new EcKeyCreateOptions("ec-key-name", hsm: true);
+Key ecKey = await Client.CreateEcKeyAsync(echsmkey);
+
+Console.WriteLine(ecKey.Name);
+Console.WriteLine(ecKey.KeyMaterial.KeyType);
+```
+
+## Troubleshooting
+
+### General
+When you interact with the Azure Key Vault Key client library using the .NET SDK, errors returned by the service correspond to the same HTTP status codes returned for [REST API][keyvault_rest] requests.
+
+For example, if you try to retrieve a Key that doesn't exist in your Key Vault, a `404` error is returned, indicating `Not Found`.
+
+```c#
+Key key = client.GetKey("some_key");
```

Comment:
You should update this to wrap in a `try/catch`, show the type of exception, and `Console.WriteLine(ex.ToString())` to help users get started.